sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py

```diff
@@ -0,0 +1,407 @@
+                **kwargs
+            )
+            return database_proxy
+        except CosmosResourceNotFoundError:
+            return await self.create_database(
+                id,
+                populate_query_metrics=populate_query_metrics,
+                offer_throughput=offer_throughput,
+                **kwargs
+            )
+
+    def get_database_client(self, database):
+        # type: (Union[str, DatabaseProxy, Dict[str, Any]]) -> DatabaseProxy
+        """Retrieve an existing database with the ID (name) `id`.
+
+        :param database: The ID (name), dict representing the properties or
+            `DatabaseProxy` instance of the database to read.
+        :type database: str or dict(str, str) or ~azure.cosmos.DatabaseProxy
+        :returns: A `DatabaseProxy` instance representing the retrieved database.
+        :rtype: ~azure.cosmos.DatabaseProxy
+        """
+        if isinstance(database, DatabaseProxy):
+            id_value = database.id
+        else:
+            try:
+                id_value = database["id"]
+            except TypeError:
+                id_value = database
+
+        return DatabaseProxy(self.client_connection, id_value)
+
+    @distributed_trace
+    def list_databases(
+        self,
+        max_item_count=None,  # type: Optional[int]
+        populate_query_metrics=None,  # type: Optional[bool]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> Iterable[Dict[str, Any]]
+        """List the databases in a Cosmos DB SQL database account.
+
+        :param int max_item_count: Max number of items to be returned in the enumeration operation.
+        :param bool populate_query_metrics: Enable returning query metrics in response headers.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :returns: An Iterable of database properties (dicts).
+        :rtype: Iterable[dict[str, str]]
+        """
+        feed_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
+        if max_item_count is not None:
+            feed_options["maxItemCount"] = max_item_count
+        if populate_query_metrics is not None:
+            feed_options["populateQueryMetrics"] = populate_query_metrics
+
+        result = self.client_connection.ReadDatabases(options=feed_options, **kwargs)
+        if response_hook:
+            response_hook(self.client_connection.last_response_headers)
+        return result
+
+    @distributed_trace
+    def query_databases(
+        self,
+        query=None,  # type: Optional[str]
+        parameters=None,  # type: Optional[List[str]]
+        enable_cross_partition_query=None,  # type: Optional[bool]
+        max_item_count=None,  # type:  Optional[int]
```

Comment:
This type is meant to be a List of Dicts, thanks for catching that.

This bit, however, seems to vary on the team's implementation for the python sdk's; storage blob and eventhub checkpointstore sdk just pass in the query and send everything else into **kwargs, table sdk sends in a filter and reads the rest from **kwargs, and others like data factory and synapse have their own models:
![image](https://user-images.githubusercontent.com/30335873/139741369-a460318e-9bf8-4a67-b5e8-fb1c9bf433cf.png)

Reply:
s

Reply:
in this case, I think it's worth the time for us to split it out of kwargs since it's information still directly pertinent to the query that will be passed in by the user (query and query params)
 
for comparison, the other keywords that are used by the method from kwargs are: initial_headers, request_options, response_hook, session_token - which relate more to configs and settings, not the query per se